### DISCUSSION TITLE
Theme switch request revalidates all other loaders. Can this be prevented?

### DISCUSSION THREAD

**Discussion by gnapse** [epicweb-dev/epic-stack#765] [Q&A] [answered]
I know about shouldRevalidate. But the way I understand it, I'd need to do that for every loader in the app, to detect if the request that was made was the theme update.

Is there a way to do this in a more global way? Preventing all loaders from revalidate when the theme switch request is sent.

**Comment by kentcdodds** [comment, marked as answer]
You could make the request using regular fetch. Or you could skip the request and just assign document.cookie directly, but then in the unlikely event any of your data depends on the theme it would not revalidate... I have a situation like this in my app though so it does happen.

**Reply by gnapse** [reply]
Good point. Though in general is uncommon (I would say) that app data changes due to the theme. Most commonly this is not the case.

Anyway, yes, setting the cookie could work for this particular case. I still have a use case of a user preference that needs to be updated in the DB, but does not affect most data. Reloading all loaders seems wasteful in this case. But I'll deal with it if/when it becomes a problem.

Thanks!
